src/overlays/actors/ovl_En_Horse/z_en_horse.h

```diff
@@ -4,50 +4,192 @@
+#define ENHORSE_INACTIVE                   (1 << 13)  /*      0x2000 */
+#define ENHORSE_OBSTACLE                   (1 << 14)  /*      0x4000 */
+#define ENHORSE_TURNING_TO_PLAYER          (1 << 15)  /*      0x8000 */
+#define ENHORSE_UNRIDEABLE                 (1 << 16)  /*    0x1 0000 */
+#define ENHORSE_CANT_JUMP                  (1 << 17)  /*    0x2 0000 */
+#define ENHORSE_FLAG_18                    (1 << 18)  /*    0x4 0000 */
+#define ENHORSE_FLAG_19                    (1 << 19)  /*    0x8 0000 */
+#define ENHORSE_FLAG_20                    (1 << 20)  /*   0x10 0000 */
+#define ENHORSE_FLAG_21                    (1 << 21)  /*   0x20 0000 */
+#define ENHORSE_FIRST_BOOST_REGEN          (1 << 22)  /*   0x40 0000 */
+#define ENHORSE_INGO_WON                   (1 << 23)  /*   0x80 0000 */
+#define ENHORSE_FLAG_24                    (1 << 24)  /*  0x100 0000 */
+#define ENHORSE_FLAG_25                    (1 << 25)  /*  0x200 0000 */
+#define ENHORSE_FLAG_26                    (1 << 26)  /*  0x400 0000 */
+#define ENHORSE_DRAW                       (1 << 27)  /*  0x800 0000 */
+#define ENHORSE_FLAG_28                    (1 << 28)  /* 0x1000 0000 */
+#define ENHORSE_FLAG_29                    (1 << 29)  /* 0x2000 0000 */
+#define ENHORSE_FLAG_30                    (1 << 30)  /* 0x4000 0000 */
+#define ENHORSE_FLAG_31                    (1 << 31)  /* 0x8000 0000 */
+
 struct EnHorse;
 
+typedef enum {
+    /* 0 */ PLAYER_DIR_FRONT_R,
+    /* 1 */ PLAYER_DIR_FRONT_L,
+    /* 2 */ PLAYER_DIR_BACK_R,
+    /* 3 */ PLAYER_DIR_BACK_L,
+    /* 4 */ PLAYER_DIR_SIDE_R,
+    /* 5 */ PLAYER_DIR_SIDE_L,
+} EnHorsePlayerDir;
+
+typedef enum {
+    /* 0 */ ENHORSE_ANIM_IDLE,
+    /* 1 */ ENHORSE_ANIM_WHINNEY,
+    /* 2 */ ENHORSE_ANIM_STOPPING,
+    /* 3 */ ENHORSE_ANIM_REARING,
+    /* 4 */ ENHORSE_ANIM_WALK,
+    /* 5 */ ENHORSE_ANIM_TROT,
+    /* 6 */ ENHORSE_ANIM_GALLOP,
+    /* 7 */ ENHORSE_ANIM_LOW_JUMP,
+    /* 8 */ ENHORSE_ANIM_HIGH_JUMP,
+} EnHorseAnimationIndex;
+
+typedef enum {
+    /* 0 */ HORSE_EPONA,
+    /* 1 */ HORSE_HNI,
+} HorseType;
+
+typedef void (*EnHorsePostdrawFunc)(struct EnHorse*, GlobalContext*);
+
 typedef struct EnHorse {
-    /* 0x000 */ Actor actor;
-    /* 0x014C */ s32 unk_14C;
-    /* 0x0150 */ char unk_150[0x008];
-    /* 0x0150 */ s32 unk_158;
-    /* 0x0150 */ char unk_15C[0x094];
-    /* 0x01F0 */ u32 unk_1F0;
-    /* 0x01F4 */ char unk_1F4[0x01C];
-    /* 0x0210 */ s32 unk_210;
-    /* 0x0214 */ f32 unk_214;
-    /* 0x0218 */ char unk_218[0x040];
-    /* 0x0258 */ Vec3f unk_258;
-    /* 0x0264 */ char unk_264[0x124];
-    /* 0x0388 */ s32 unk_388;
-    /* 0x038C */ char unk_38C[0x14];
-    /* 0x03A0 */ s32 unk_3A0;
-    /* 0x03A4 */ char unk_3A4[0x58];
+    /* 0x0000 */ Actor actor;
+    /* 0x014C */ EnHorseAction action;
+    /* 0x0150 */ s32 noInputTimer;
+    /* 0x0154 */ s32 noInputTimerMax;
+    /* 0x0158 */ s32 type;
+    /* 0x015C */ s8 bankIndex;
+    /* 0x0160 */ PSkinAwb skin;
+    /* 0x01F0 */ u32 flags;
```

Comment:
`flags` is already a field of the base actor struct, so we may want to name this something slightly different to reduce confusion. even something like `horseFlags`? dunno

Reply:
I agree we should avoid using field names that are already in the base struct. Maybe a more specific name could be used if it makes sense, like `stateFlags` or `actionFlags`?